diff --git a/build/boards/cv180x/cv1800b_milkv_duo_musl_riscv64_sd/dts_riscv/cv1800b_milkv_duo_musl_riscv64_sd.dts b/build/boards/cv180x/cv1800b_milkv_duo_musl_riscv64_sd/dts_riscv/cv1800b_milkv_duo_musl_riscv64_sd.dts
index 9d840d088..4e637999b 100644
--- a/build/boards/cv180x/cv1800b_milkv_duo_musl_riscv64_sd/dts_riscv/cv1800b_milkv_duo_musl_riscv64_sd.dts
+++ b/build/boards/cv180x/cv1800b_milkv_duo_musl_riscv64_sd/dts_riscv/cv1800b_milkv_duo_musl_riscv64_sd.dts
@@ -10,9 +10,23 @@
 
 &spi2 {
 	status = "okay";
-
-	spidev@0 {
+	/delete-node/ spidev@0;
+	st7789v: st7789v@0{
+		compatible = "sitronix,st7789v";
+		reg = <0>;
 		status = "okay";
+		spi-max-frequency = <48000000>;
+		spi-cpol;
+		spi-cpha;
+		rotate = <0>;
+		fps = <60>;
+		rgb;
+		buswidth = <8>;
+
+		dc = <&porta 24 GPIO_ACTIVE_HIGH>;
+		reset = <&porta 23 GPIO_ACTIVE_HIGH>;
+
+		debug = <0x0>;
 	};
 };
 
diff --git a/build/boards/cv181x/sg2002_milkv_duo256m_musl_riscv64_sd/dts_riscv/sg2002_milkv_duo256m_musl_riscv64_sd.dts b/build/boards/cv181x/sg2002_milkv_duo256m_musl_riscv64_sd/dts_riscv/sg2002_milkv_duo256m_musl_riscv64_sd.dts
index c8332b316..84ae70b51 100644
--- a/build/boards/cv181x/sg2002_milkv_duo256m_musl_riscv64_sd/dts_riscv/sg2002_milkv_duo256m_musl_riscv64_sd.dts
+++ b/build/boards/cv181x/sg2002_milkv_duo256m_musl_riscv64_sd/dts_riscv/sg2002_milkv_duo256m_musl_riscv64_sd.dts
@@ -16,9 +16,24 @@
 
 &spi2 {
 	status = "okay";
+	/delete-node/ spidev@0;
 
-	spidev@0 {
+	st7789v: st7789v@0{
+		compatible = "sitronix,st7789v";
+		reg = <0>;
 		status = "okay";
+		spi-max-frequency = <48000000>;
+		spi-cpol;
+		spi-cpha;
+		rotate = <0>;
+		fps = <60>;
+		rgb;
+		buswidth = <8>;
+
+		dc = <&porta 24 GPIO_ACTIVE_HIGH>;
+		reset = <&porta 23 GPIO_ACTIVE_HIGH>;
+
+		debug = <0x0>;
 	};
 };
 
diff --git a/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_emmc/dts_riscv/sg2000_milkv_duos_musl_riscv64_emmc.dts b/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_emmc/dts_riscv/sg2000_milkv_duos_musl_riscv64_emmc.dts
index aeca0bfcc..5c209399b 100644
--- a/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_emmc/dts_riscv/sg2000_milkv_duos_musl_riscv64_emmc.dts
+++ b/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_emmc/dts_riscv/sg2000_milkv_duos_musl_riscv64_emmc.dts
@@ -43,9 +43,24 @@
 
 &spi3 {
 	status = "okay";
+	/delete-node/ spidev@0;
 
-	spidev@0 {
+	st7789v: st7789v@0{
+		compatible = "sitronix,st7789v";
+		reg = <0>;
 		status = "okay";
+		spi-max-frequency = <48000000>;
+		spi-cpol;
+		spi-cpha;
+		rotate = <0>;
+		fps = <60>;
+		rgb;
+		buswidth = <8>;
+
+		dc = <&porta 18 GPIO_ACTIVE_HIGH>;
+		reset = <&porta 28 GPIO_ACTIVE_HIGH>;
+
+		debug = <0x0>;
 	};
 };
 
diff --git a/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_sd/dts_riscv/sg2000_milkv_duos_musl_riscv64_sd.dts b/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_sd/dts_riscv/sg2000_milkv_duos_musl_riscv64_sd.dts
index 1ea1d4632..c76948a74 100644
--- a/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_sd/dts_riscv/sg2000_milkv_duos_musl_riscv64_sd.dts
+++ b/build/boards/cv181x/sg2000_milkv_duos_musl_riscv64_sd/dts_riscv/sg2000_milkv_duos_musl_riscv64_sd.dts
@@ -43,9 +43,24 @@
 
 &spi3 {
 	status = "okay";
+	/delete-node/ spidev@0;
 
-	spidev@0 {
+	st7789v: st7789v@0{
+		compatible = "sitronix,st7789v";
+		reg = <0>;
 		status = "okay";
+		spi-max-frequency = <48000000>;
+		spi-cpol;
+		spi-cpha;
+		rotate = <0>;
+		fps = <60>;
+		rgb;
+		buswidth = <8>;
+
+		dc = <&porta 18 GPIO_ACTIVE_HIGH>;
+		reset = <&porta 28 GPIO_ACTIVE_HIGH>;
+
+		debug = <0x0>;
 	};
 };

static void fbtft_update_display(struct fbtft_par *par, unsigned int start_line,
